### PR TITLE
ID-3220 [FIX] Ensure pie chart appears correctly after applying changes in component appearance settings

### PR DESCRIPTION
**Pull request by NikitaJadhavXoriant**
**Product areas affected**

Studio -> Pie Chart -> Theme appearance settings

**What does this PR do?**

Implemented code, so pie chart appears correctly after applying changes in component appearance settings

**JIRA ticket**

[ID-3220](https://weboo.atlassian.net/browse/ID-3220)

**Result**

https://github.com/Fliplet/fliplet-widget-chart-pie/assets/108272606/da12ec50-1671-4e6c-98f7-cc428c3298f6

**Checklist**

None

**Testing instructions**

None

**Deployment instructions**

None

**Author concerns**

None

[ID-3220]: https://weboo.atlassian.net/browse/ID-3220?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: js/build.js [hunk 1]
```diff
@@ -10,9 +10,16 @@ Fliplet.Widget.instance('chart-pie-1-1-0', function(data) {
   if (themeInstance) {
     Object.assign({}, themeInstance.data.values);
 
+    if(!themeValues) {
+      var themeValue = themeInstance.data.values;
```
Comment (tonytlwu) [inline js/build.js]: Declaration of the variable should occur before the `if`. Otherwise from the way the code reads, there's no way `themeValues` would have a value.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/build.js (hunk 1 above)
Comment (NikitaJadhavXoriant) [reply js/build.js]: @tonytlwu, declaration of `themeValues` is already at [this](https://github.com/Fliplet/fliplet-widget-chart-pie/blob/master/js/build.js#L8) line before `if` condition.

Please provide your inputs on this.

Comment (tonytlwu) [reply js/build.js]: Then you don't need the `var` bit on this line

Comment (NikitaJadhavXoriant) [reply js/build.js]: @tonytlwu, both are different variables, `themeValues` and `themeValue`. And in `if `condition we are checking `themeValues` which we declared previously.

Comment (tonytlwu) [reply js/build.js]: @NikitaJadhavXoriant I see. I didn't spot the difference with the `s`. In that case this is fine.

File: js/build.js [hunk 2]
```diff
@@ -10,9 +10,16 @@ Fliplet.Widget.instance('chart-pie-1-1-0', function(data) {
   if (themeInstance) {
     Object.assign({}, themeInstance.data.values);
 
+    if(!themeValues) {
```
Comment (tonytlwu) [inline js/build.js]: Isn't this always undefined if it is only declared as `undefined` 5 lines above?

Comment (NikitaJadhavXoriant) [reply js/build.js]: @tonytlwu, first time it's undefined that is why assigning values [here](https://github.com/NikitaJadhavXoriant/fliplet-widget-chart-pie/blob/%40nikitaxor/FIX-ID-3220/js/build.js#L17) but once [setThemeValues()](https://github.com/Fliplet/fliplet-widget-chart-pie/blob/master/js/build.js#L294) method called, then we are getting themeValues.

File: js/build.js [hunk 3]
```diff
@@ -10,9 +10,14 @@ Fliplet.Widget.instance('chart-pie-1-1-0', function(data) {
...
 
+    var themeValue = themeInstance.data.values;
+    var widgetValue = getColors(themeInstance.data.widgetInstances);
+
+    themeValues = Object.assign(themeValue, widgetValue);
+
     _.forEach(themeInstance.data.widgetInstances, function(widgetProp) {
-      if (chartId === widgetProp.id) {
-        Object.assign(themeValues, widgetProp.values);
+      if (chartId === widgetProp.id && themeValues) {
```
Comment (tonytlwu) [inline js/build.js]: Why is this part necessary? I imagine after `Object.assign()` in the line above, it will always be truthy.

```suggestion
      if (chartId === widgetProp.id) {
```

If not, please give me the full scenario of when `themeValues` would be falsey.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/build.js [hunk 4]
```diff
@@ -10,9 +10,14 @@ Fliplet.Widget.instance('chart-pie-1-1-0', function(data) {
   if (themeInstance) {
     Object.assign({}, themeInstance.data.values);
 
+    var themeValue = themeInstance.data.values;
+    var widgetValue = getColors(themeInstance.data.widgetInstances);
+
+    themeValues = Object.assign(themeValue, widgetValue);
+
     _.forEach(themeInstance.data.widgetInstances, function(widgetProp) {
```
Comment (tonytlwu) [inline js/build.js]: I realised that `themeInstance.data.widgetInstances` could be quite large because it has all the widget instances across all the screens. Let's shortcut this. Assuming you get the code right, we can skip QA with Kathi to save time.

```js
    _.some(themeInstance.data.widgetInstances, function(widgetProp) {
      if (chartId === widgetProp.id) {
        Object.assign(widgetProp.values, themeValues);

        return true;
      }
    });
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Review by tonytlwu — APPROVED** [review]
